array.go

```diff
@@ -66,6 +66,15 @@ func (a *Array) Raw() []interface{} {
 	return a.value
 }
 
+func (a *Array) Decode(target *[]interface{}) {
+	a.chain.enter("Decode")
+	defer a.chain.leave()
+
+	if decode := canonDecode(a.chain, a.value, target); !decode {
+		a.chain.failed()
```

Comment:
No need for this call (check its source code). Actually seems we don't need to check return value of canonDecode at all, and it doesn't need to return anything